### ISSUE TITLE
When will Sourcery support Apple M1 Silicon? 

### ISSUE THREAD

**Issue by pzelnip** [sourcery-ai/sourcery#121]
### Issue description or question

I use Sourcery via the VS Code extension, and while VS Code is now native M1, I noticed that Sourcery still appears in Activity Monitor as an Intel binary requiring Rosetta.  Wondering if/when Sourcery will have a M1 native version?

### Sourcery Version

VS Code extension version 0.8.7

### Code editor or IDE name and version

VS Code

### OS name and version

OSX

**Comment by Hellebore** [comment]
Sadly this isn't a priority (we also don't have an M1 device to test on) so can't give an answer at the moment.

**Comment by songmeixu** [comment]
Hi @pzelnip, could you use sourcery normally except it isn't native supported? As my output of this ext shows an error:
```log
[Error - 12:50:17 PM] Server initialization failed.
  Message: ValueError: RecordCollection contained more than one row. Expects only one row when using RecordCollection.one
  Code: -32602 
[object Object]
```

**Comment by pzelnip** [comment]
Seemed to work normally for me, but admittedly until native M1 arrives I have sourcery disabled now in VS Code to save battery.

**Comment by Hellebore** [comment]
@songmeixu could you try again with version 0.8.8 and let me know if it works? (we've fixed a similar problem in that version) If not it would be great if you could raise another issue with your details so we can get it sorted out.

**Comment by Hellebore** [comment]
@pzelnip Ah I see that's a shame. Will leave this issue open so you can see when we do get native M1.

**Comment by pzelnip** [comment]
> Sadly this isn't a priority (we also don't have an M1 device to test on) so can't give an answer at the moment.

I don't know why it didn't didn't occur to me to mention this when I first opened the issue, but FWIW I'd absolutely be happy to help with testing Sourcery on my M1-powered Macbook Air (even if not a native M1 version, if there's metrics or data you'd like to collect around how Sourcery runs under Rosetta I'd be willing to help if there's a way I can).

I eventually caved and reinstalled Sourcery on my M1 as I found myself missing the refactorings.  I am still a bit worried about the impact on battery life but so far that seems negligible (I did a really unscientific experiment of leaving a VS Code window open with Sourcery installed for about a half hour, made a note of how much the battery drained, then repeated with it not installed and the % drain was about the same).

**Comment by jalaziz** [comment]
Came here noticing the same thing when using Sourcery with PyCharm.

Has there been any update since March? Happy to help test.

Depending on how Sourcery is built, hopefully it's a simple matter of enabling cross compilation.

**Comment by noirchen** [comment]
Any updates on the silicon version?

**Comment by diceroll123** [comment]
Works on M1-Pro MBP here.

**Comment by paddyroddy** [comment]
I clearly haven't tried since I first got an M1 but prompting @diceroll123's comment I've just had another play and the `vscode` extension is working for me on standard M1 chip 😄

**Comment by pzelnip** [comment]
@paddyroddy the current version works on M1, but is not M1-native -- ie you need Rosetta installed to run it as it's an Intel process, which you can see in Activity Monitor:
![Screen Shot 2022-07-25 at 11 27 55 AM](https://user-images.githubusercontent.com/414933/180848490-a1f09ad0-9dc4-4757-81cf-b5012bf41fea.png)

The idea of this request is to have a M1-native version of Sourcery so as to not require Rosetta/be more efficient.

**Comment by paddyroddy** [comment]
Yeah, I get you @pzelnip, but when I first tried (a while ago now) it didn't work even with Rosetta - which is why I've been following this issue

**Comment by pzelnip** [comment]
Any update on this? We're now multiple years since the launch of Apple Silicon, so it's getting harder and harder to justify running something in Rosetta.

**Comment by ryanmerolle** [comment]
I cannot use Rosetta if VS Code uses a devcontainer that is native arm.

**Comment by gquittet** [comment]
Hello guys, any update on this? 👀

**Comment by TheYarin** [comment]
I need this, too! I'm using a dev container in VSCode that is native arm.

**Comment by cmachadomarcos** [comment]
Hello, is there any update on this?

**Comment by chenhuiapp** [comment]
Need update on this. Please!

**Comment by bill-tucker-ingenovishealth** [comment]
I was able to install Rosetta on my m2 mac and log into Sourcery plug-in.  Worked great.

**Comment by ryanmerolle** [comment]
> I was able to install Rosetta on my m2 mac and log into Sourcery plug-in. Worked great.

Yes we all can.  If you run a devcontainer or docker and it defaults to the normal arm version of a container it does not work because Rosetta cannot run in a vm or container.  You can run the container using Rosetta but containers and vms in Rosetta is unbearably slow and resource intensive.  Hence the request for native arm support.  It would benefit all arm users not just Apple silicon users.

**Comment by gy-mate** [comment]
@Hellebore Is there any ETA on this? I really miss using Sourcery...
